src/main/java/io/reactivex/FlowableTransformer.java

```diff
@@ -15,14 +15,12 @@
 
 import org.reactivestreams.Publisher;
 
-import io.reactivex.functions.Function;
-
 /**
  * Interface to compose Flowables.
  *
  * @param <Upstream> the upstream value type
  * @param <Downstream> the downstream value type
  */
-public interface FlowableTransformer<Upstream, Downstream> extends Function<Flowable<Upstream>, Publisher<? extends Downstream>> {
-
+public interface FlowableTransformer<Upstream, Downstream> {
+    Publisher<? extends Downstream> apply(Flowable<Upstream> flowable);
```

Comment:
I am not certain whether the ? extends should be there or not. I know that I have long had generic type problems with compose and RxLifecycle and am not sure whether the fault is RxJava or RxLifecycle. And the results vary between Java7, Java 8, and Kotlin regarding this.

See this discussion in RxLifecycle: https://github.com/trello/RxLifecycle/issues/58